sources/zafira-services/src/main/java/com/qaprosoft/zafira/services/services/application/TestRunService.java

```diff
@@ -517,19 +517,22 @@ public String sendTestRunResultsNotification(final TestRun testRun,
 
     @Transactional(readOnly = true)
     public String exportTestRunHTML(final String id) {
```

Comment:
My understanding is that if test run is not found we'll return `null` to the caller. It is hard to evaluate if we can easily change that behavior but let's agree to avoid nullable APIs from now on. All new APIs should either return optional and somehow be designed in a fashion when default value will be returned. That kind of inter-service communication looks like a design smell to me.